azDevOps/azure/templates/v2/steps/deploy-k8s-app-kubectl-generic.yml

```diff
@@ -21,24 +22,44 @@ steps:
       login_azure: ${{ parameters.login_azure }}
       login_kubernetes: ${{ parameters.login_kubernetes }}
 
+  - bash: apt install jq
+    displayName: install jq
+
   - ${{ each deployment in parameters.deployment_defs }}:
       - bash: |
-          kubectl apply -f ${{ deployment.file }} ${{ deployment.context }}
+          set -eou pipefail
+          timeoutMinutes=10
```

Comment:
This should be passed in as configurable, and a default set. Also the deployment name might want to be passed in.

Please look at https://github.com/amido/stacks-pipeline-templates/blob/feature/cycle4/azDevOps/azure/templates/v2/steps/deploy-k8s-app-kubectl.yml where this is already implemented.

Reply:
https://github.com/amido/stacks-pipeline-templates/blob/d7f07ca59f5ecbb18775689c397bb5766da7108c/azDevOps/azure/templates/v2/steps/deploy-k8s-app-kubectl.yml#L63-L69

Reply:
Now that I look at it I notice this isn't looping for all deploys and only takes in a single deploy string, that's a bit of an oversight haha.

Reply:
For completeness this is how it's currently called in stacks-dotnet for example: https://github.com/amido/stacks-dotnet/blob/c84fa612ed91c72201a062aa01f51f3932116d8c/build/azDevOps/azure/azure-pipelines-netcore-k8s.yml#L365-L370

Reply:
Maybe we could work something out that works for all eventualities?